### ISSUE TITLE
[Bug] Timeout Provider hangs during sysprep on Window Server 2016

### ISSUE THREAD

**Issue by Techcadia** [hashicorp/packer#7735]
I originally had Teamcity setup to kill packer after 45 minutes as Sysprep hangs most of the time. After reviewing the newer releases I found that timeout was added. 

Great lets see if this works the way I would expect.

I set a 10 minute timeout on the provisioner that handled the sysprep. At this point I have seen the timeout fire but it still is hanging.

```
[20:36:26]	[Step 1/1] ==> amazon-ebs: Provisioning with powershell script: /opt/TeamCityAgent/temp/buildTmp/powershell-provisioner553676562
[20:36:29]	[Step 1/1]     amazon-ebs: Cleaning up Apps
[20:36:30]	[Step 1/1]     amazon-ebs: Running Finalization
[20:36:31]	[Step 1/1]     amazon-ebs:
[20:36:31]	[Step 1/1]     amazon-ebs: TaskPath                                       TaskName                          State
[20:36:31]	[Step 1/1]     amazon-ebs: --------                                       --------                          -----
[20:36:31]	[Step 1/1]     amazon-ebs: \                                              Amazon Ec2 Launch - Instance I... Ready
[20:36:31]	[Step 1/1]     amazon-ebs: Running Sysprep
[20:46:26]	[Step 1/1] ==> amazon-ebs: Cancelling provisioner after a timeout...
# BELOW THIS POINT IS WHERE I TOOK ACTION
[23:08:47]	[Step 1/1] Process exited with code 137
[23:08:47]	[Step 1/1] Process exited with code 137 (Step: Build (Command Line))
[23:08:47]	[Step 1/1] Step Build (Command Line) interrupted
[23:08:47]	Build was interrupted. Artifacts will not be published for this build
[23:08:47]	Build canceled
```

relevant information
Packer 1.4.1
AWS ELB Provider
Window Server 2016 - ami-0ba970d2fe7bbb7cf (us-west-2)
Using Packer, Chef and Powershell for Provisioning

Somehow I get sysprep into a bad state where a popup box gets thrown saying to check the log.

This can be repoed without sysprep using the following provisioner
```
"provisioners": [
  {
    "type": "powershell",
    "inline": [
      "Add-Type -AssemblyName PresentationCore,PresentationFramework",
      "[System.Windows.MessageBox]::Show('Message','Title',[System.Windows.MessageBoxButton]::Ok,[System.Windows.MessageBoxImage]::Error)",
      "Write-Host 'We Should never reach this point'"
    ],
    "timeout": "45s"
  }
]
```

NOTE: I would assume the timeout should trigger a job failure. But that does not seem to be the case if I test the timeout in a vacuum it just skips the provisioner.

**Comment by azr** [comment]
Hello there, thanks for reporting. That's annoying ! I'm looking into it, this should return properly on timeout ( context cancellation ).

the following snippet returned:
```go
package main

import (
	"context"
	"fmt"
	"os"
	"os/exec"
	"time"
)

func powershell(ctx context.Context, arg ...string) {
	cmd := exec.CommandContext(ctx, "powershell", arg...)
	cmd.Stdout = os.Stdout
	cmd.Stderr = os.Stderr

	err := cmd.Run()
	fmt.Printf("err: %s.\n", err)
}

func main() {
	ctx, _ := context.WithTimeout(context.Background(), time.Second*3)
	powershell(ctx, "Add-Type -AssemblyName PresentationCore,PresentationFramework,System.Windows.Forms ;", "[System.Windows.MessageBox]::Show('Message','Title',[System.Windows.MessageBoxButton]::Ok,[System.Windows.MessageBoxImage]::Error)")
}

```

**Comment by Techcadia** [comment]
@azr Downloaded master and compiled it. This looks like its solved with ticket #7657 or #7705 not sure which one solves it but I can confirm that is does bail out now successfully.

On a side note I am not sure if we really want to defer f.Close() and also f.Close()

```
diff --git a/provisioner/powershell/provisioner.go b/provisioner/powershell/provisioner.go
index 6b0038947..64f89dd4d 100644
--- a/provisioner/powershell/provisioner.go
+++ b/provisioner/powershell/provisioner.go
@@ -292,9 +292,6 @@ func (p *Provisioner) Provision(ctx context.Context, ui packer.Ui, comm packer.C
                        return err
                }

-               // Close the original file since we copied it
-               f.Close()
-
                log.Printf("%s returned with exit code %d", p.config.RemotePath, cmd.ExitStatus())

                if err := p.config.ValidExitCode(cmd.ExitStatus()); err != nil {
```

**Comment by azr** [comment]
Hey @Techcadia awesome, thanks ! I think it's #7657.

On the double file close, this should not cause an issue, but code quality is always a concert ! Please open a PR with these change; I'd happily merge.
